### PR TITLE
DefaultProviderFirstParams add THREADPOOL_KEY

### LINKED ISSUE #13635: Low version of consumer reports error when dubbo3.3provider uses virtual threading
- [ ] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: 3.3 provider  3.2consumer
* Operating System version: xxx
* Java version: jdk21

### Steps to reproduce this issue

1.  provider of dubbo 3.3 uses'dubbo.protocol.threadpool = virtual 'and start
2. dubbo3.2 consumer start
3. The exception log will be output directly

Pls. provide [GitHub address] to reproduce this issue.

### Actual Behavior

If there is an exception, please attach the exception trace:

```
2024-01-10 09:35:40.709 ERROR 59497 --- [           main] o.a.d.r.integration.RegistryDirectory    :  [DUBBO] urls to invokers error, dubbo version: 3.2.10, current host: 10.57.240.121, error code: 3-1. This may be caused by inconsistency between the client protocol and the protocol of the server, go to https://dubbo.apache.org/faq/3/1 to find instructions. 

java.lang.IllegalStateException: urls to invokers error. invokerUrls.size :1, invoker.size :0. urls :[dubbo://10.57.17.39:20880/cn.tongdun.app.service.TestService?application=java-demo-002169&background=false&category=providers,configurators,routers&check=false&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&file-cache=true&generic=false&interface=cn.tongdun.app.service.TestService&methods=test&pid=59497&prefer.serialization=fastjson2,hessian2&register-mode=all&release=3.2.10&revision=1.0.1&service-name-mapping=true&side=provider&sticky=false&threadpool=virtual&unloadClusterRelated=false&version=1.0.0]
	at org.apache.dubbo.registry.integration.RegistryDirectory.refreshInvoker(RegistryDirectory.java:341)
	at org.apache.dubbo.registry.integration.RegistryDirectory.refreshOverrideAndInvoker(RegistryDirectory.java:254)
	at org.apache.dubbo.registry.integration.RegistryDirectory.notify(RegistryDirectory.java:230)
	at org.apache.dubbo.registry.support.AbstractRegistry.notify(AbstractRegistry.java:574)
	at org.apache.dubbo.registry.support.FailbackRegistry.doNotify(FailbackRegistry.java:421)
	at org.apache.dubbo.registry.support.FailbackRegistry.notify(FailbackRegistry.java:408)
	at org.apache.dubbo.registry.zookeeper.ZookeeperRegistry.doSubscribe(ZookeeperRegistry.java:268)
	at org.apache.dubbo.registry.support.FailbackRegistry.subscribe(FailbackRegistry.java:324)
	at org.apache.dubbo.registry.ListenerRegistryWrapper.subscribe(ListenerRegistryWrapper.java:87)
	at org.apache.dubbo.registry.integration.DynamicDirectory.subscribe(DynamicDirectory.java:186)
	at org.apache.dubbo.registry.integration.RegistryDirectory.lambda$subscribe$0(RegistryDirectory.java:147)
	at org.apache.dubbo.metrics.event.MetricsEventBus.post(MetricsEventBus.java:76)
	at org.apache.dubbo.metrics.event.MetricsEventBus.post(MetricsEventBus.java:59)
	at org.apache.dubbo.registry.integration.RegistryDirectory.subscribe(RegistryDirectory.java:146)
	at org.apache.dubbo.registry.integration.RegistryProtocol.doCreateInvoker(RegistryProtocol.java:695)
	at org.apache.dubbo.registry.integration.InterfaceCompatibleRegistryProtocol.getInvoker(InterfaceCompatibleRegistryProtocol.java:58)
	at org.apache.dubbo.registry.client.migration.MigrationInvoker.refreshInterfaceInvoker(MigrationInvoker.java:482)
	at org.apache.dubbo.registry.client.migration.MigrationInvoker.migrateToApplicationFirstInvoker(MigrationInvoker.java:255)
	at org.apache.dubbo.registry.client.migration.MigrationRuleHandler.refreshInvoker(MigrationRuleHandler.java:78)
	at org.apache.dubbo.registry.client.migration.MigrationRuleHandler.doMigrate(MigrationRuleHandler.java:62)
	at org.apache.dubbo.registry.client.migration.MigrationRuleListener.onRefer(MigrationRuleListener.java:285)
	at org.apache.dubbo.registry.integration.RegistryProtocol.interceptInvoker(RegistryProtocol.java:658)
	at org.apache.dubbo.registry.integration.RegistryProtocol.doRefer(RegistryProtocol.java:622)
	at org.apache.dubbo.registry.integration.RegistryProtocol.refer(RegistryProtocol.java:603)
	at org.apache.dubbo.qos.protocol.QosProtocolWrapper.refer(QosProtocolWrapper.java:85)
	at org.apache.dubbo.rpc.protocol.ProtocolListenerWrapper.refer(ProtocolListenerWrapper.java:84)
	at org.apache.dubbo.rpc.protocol.ProtocolSecurityWrapper.refer(ProtocolSecurityWrapper.java:112)
	at org.apache.dubbo.rpc.cluster.filter.ProtocolFilterWrapper.refer(ProtocolFilterWrapper.java:72)
	at org.apache.dubbo.rpc.protocol.ProtocolSerializationWrapper.refer(ProtocolSerializationWrapper.java:55)
	at org.apache.dubbo.rpc.protocol.InvokerCountWrapper.refer(InvokerCountWrapper.java:48)
	at org.apache.dubbo.rpc.Protocol$Adaptive.refer(Protocol$Adaptive.java)
	at org.apache.dubbo.config.ReferenceConfig.createInvoker(ReferenceConfig.java:656)
	at org.apache.dubbo.config.ReferenceConfig.createProxy(ReferenceConfig.java:486)
	at org.apache.dubbo.config.ReferenceConfig.init(ReferenceConfig.java:369)
	at org.apache.dubbo.config.ReferenceConfig.get(ReferenceConfig.java:238)
	at org.apache.dubbo.config.utils.SimpleReferenceCache.get(SimpleReferenceCache.java:140)
	at org.apache.dubbo.config.deploy.DefaultModuleDeployer.lambda$referServices$6(DefaultModuleDeployer.java:542)
	at java.base/java.util.concurrent.ConcurrentHashMap$ValuesView.forEach(ConcurrentHashMap.java:4783)
	at org.apache.dubbo.config.deploy.DefaultModuleDeployer.referServices(DefaultModuleDeployer.java:514)
	at org.apache.dubbo.config.deploy.DefaultModuleDeployer.startSync(DefaultModuleDeployer.java:183)
	at org.apache.dubbo.config.deploy.DefaultModuleDeployer.start(DefaultModuleDeployer.java:156)
	at org.apache.dubbo.config.spring.context.DubboDeployApplicationListener.onContextRefreshedEvent(DubboDeployApplicationListener.java:157)
	at org.apache.dubbo.config.spring.context.DubboDeployApplicationListener.onApplicationEvent(DubboDeployApplicationListener.java:143)
	at org.apache.dubbo.config.spring.context.DubboDeployApplicationListener.onApplicationEvent(DubboDeployApplicationListener.java:52)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:178)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:171)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:145)
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:429)
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:386)
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:949)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:594)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:147)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:732)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:409)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:308)
	at cn.tongdun.app.AppMain.main(AppMain.java:133)
```

### PR DESCRIPTION
**Pull request by funky-eyes**
## What is the purpose of the change
fixes #13635 
3.2.10:
```
2024-01-11 15:57:48.780 ERROR 34897 --- [           main] o.a.d.r.integration.RegistryDirectory    :  [DUBBO] urls to invokers error, dubbo version: 3.2.10, current host: 10.57.240.121, error code: 3-1. This may be caused by inconsistency between the client protocol and the protocol of the server, go to https://dubbo.apache.org/faq/3/1 to find instructions. 

java.lang.IllegalStateException: urls to invokers error. invokerUrls.size :1, invoker.size :0. urls :[dubbo://10.57.240.121:20880/cn.app.service.TestService?application=java-demo&background=false&category=providers,configurators,routers&check=false&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&file-cache=true&generic=false&interface=cn.app.service.TestService&methods=test&pid=34897&prefer.serialization=fastjson2,hessian2&qos.port=23333&register-mode=all&release=3.2.10&revision=1.0.0&service-name-mapping=true&side=provider&sticky=false&threadpool=virtual&unloadClusterRelated=false&version=1.0.0]
```
After this pr change:
```
2024-01-11 16:00:49.028  INFO 35191 --- [           main] .r.c.e.l.ServiceInstancesChangedListener :  [DUBBO] Notify service cn.app.service.TestService:1.0.0:null with urls 1, dubbo version: 3.2.10, current host: 10.57.240.121
2024-01-11 16:00:49.029  INFO 35191 --- [           main] .d.r.c.ServiceDiscoveryRegistryDirectory :  [DUBBO] Refreshed invoker size 1 from registry ServiceDiscoveryRegistryDirectory(registry: zookeeper-main6.te.td:2181,zookeeper-main7.te.td:2181,zookeeper-main8.te.td:2181, subscribed key: [java-demo])-Directory(invokers: 1[10.57.240.121:20880], validInvokers: 1[10.57.240.121:20880], invokersToReconnect: 0[]), dubbo version: 3.2.10, current host: 10.57.240.121
2024-01-11 16:00:49.029  INFO 35191 --- [           main] .d.r.c.ServiceDiscoveryRegistryDirectory :  [DUBBO] Received invokers changed event from registry. Registry type: instance. Service Key: cn.app.service.TestService:1.0.0. Urls Size : 1. Invokers Size : 1. Available Size: 1. Available Invokers : 10.57.240.121:20880, dubbo version: 3.2.10, current host: 10.57.240.121
2024-01-11 16:00:49.040  INFO 35191 --- [           main] .r.c.m.DefaultMigrationAddressComparator :  [DUBBO] serviceKey:cn.app.service.TestService:1.0.0 Instance address size 1, interface address size 1, threshold 0.0, dubbo version: 3.2.10, current host: 10.57.240.121
2024-01-11 16:00:49.041  INFO 35191 --- [           main] o.a.d.r.c.m.MigrationRuleHandler         :  [DUBBO] Succeed Migrated to APPLICATION_FIRST mode. Service Name: cn.app.service.TestService:1.0.0, dubbo version: 3.2.10, current host: 10.57.240.121
2024-01-11 16:00:49.041  INFO 35191 --- [           main] org.apache.dubbo.config.ReferenceConfig  :  [DUBBO] Referred dubbo service: [cn.app.service.TestService]. it's not GenericService reference, dubbo version: 3.2.10, current host: 10.57.240.121

```

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/support/DefaultProviderFirstParams.java [hunk 1]
```diff
@@ -27,12 +27,13 @@
...
 import static org.apache.dubbo.common.constants.CommonConstants.RELEASE_KEY;
 import static org.apache.dubbo.common.constants.CommonConstants.TAG_KEY;
+import static org.apache.dubbo.common.constants.CommonConstants.THREADPOOL_KEY;
 import static org.apache.dubbo.common.constants.CommonConstants.TIMESTAMP_KEY;
 
 public class DefaultProviderFirstParams implements ProviderFirstParams {
     private static final Set<String> PARAMS = Collections.unmodifiableSet(new HashSet<String>() {
         {
-            addAll(Arrays.asList(RELEASE_KEY, DUBBO_VERSION_KEY, METHODS_KEY, TIMESTAMP_KEY, TAG_KEY));
+            addAll(Arrays.asList(RELEASE_KEY, DUBBO_VERSION_KEY, METHODS_KEY, TIMESTAMP_KEY, TAG_KEY, THREADPOOL_KEY));
```
Comment (AlbumenJ) [inline dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/support/DefaultProviderFirstParams.java]: Is it possible to also add some useless keys like `executor-management-mode`?

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/support/DefaultProviderFirstParams.java (hunk 1 above)
Comment (funky-eyes) [reply dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/support/DefaultProviderFirstParams.java]: > Is it possible to also add some useless keys like `executor-management-mode`?

ok

Comment (funky-eyes) [reply dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/support/DefaultProviderFirstParams.java]: done

**Review by kimmking — APPROVED** [review]

**Comment by AlbumenJ** [comment]
- @funkye Can you please submit apache/dubbo#13637 again to 3.3.x branch?
- In the new PR, we should sort out all the keys that have been registered in the registration center, and ensure that there will be no new useless keys through the blacklist mechanism and day.

**Pull request merged**